db/migrate/20150404193023_participation_counter.rb

```diff
@@ -0,0 +1,23 @@
+class ParticipationCounter < ActiveRecord::Migration
+  def up
+    add_column :participations, :count, :int, :null => false, default: 1
+    execute "update participations
+      set count =
+          (
+              (select count(l.id)
+               from likes l
+               where
+                  l.target_id = participations.target_id
+                  and l.author_id = participations.author_id) +
+              (select count(c.id)
+               from comments c
+               where
+                  c.commentable_id = participations.target_id
+                  and c.author_id = participations.author_id)
+          )"
```

Comment:
What I can come up with is not really worth it:

```ruby
likes_count = Like.select("COUNT(likes.id)")
                  .where("likes.target_id = participations.target_id")
                  .to_sql
comments_count = Comment.select("COUNT(comments.id)")
                        .where("comments.commentable_id = participations.target_id")
                        .where("comments.author_id = participations.author_id")
                        .to_sql
Participation.update_all("count = (#{likes_count}) + (#{comments_count)")
```

So maybe just capitalize the SQL keywords to make it a bit more readable.